lib/ruote/postgres/storage.rb

```diff
@@ -297,14 +298,30 @@ def has_json?
         server_version[0] >= 9 && server_version[1] >= 2
       end
 
+      def reconnect
+        @pg = db_connect
+      end
+
+      def db_connect
+        PGconn.connect_start(@pg.host, @pg.port, @pg.options, @pg.tty, @pg.db, @pg.user, @pg.pass)
+      end
+
       def safe_pg(&block)
+        retries ||= 0
+
         @mutex.synchronize do
           yield
         end
       rescue *CONNECTION_ERRORS => e
+        if @retry_on_connection_error && retries < 1
```

Comment:
So will retry only 1 time? Can be better to set a max_retries param to retry N times?